File: baekjoon/bfs/1697/1697.java
import java.util.*;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main {
  static int[] visited;
  static int start,end;

  public static int bfs() {
    Queue<Integer> queue = new LinkedList<Integer>();
    int pop;
    queue.offer(start);
    visited[start] = 0;
    while(true){
      pop = queue.poll();
      if(pop == end) return visited[pop];
      if(pop+1 <= 100000 && visited[pop+1] == 0){
        queue.offer(pop+1);
        visited[pop+1] = visited[pop] + 1;
      }
      if(pop-1 >= 0 && visited[pop-1] == 0){
        queue.offer(pop-1);
        visited[pop-1] = visited[pop] + 1;
      }
      if(pop*2 <= 100000 && visited[pop*2] == 0){
        queue.offer(pop*2);
        visited[pop*2] = visited[pop] + 1;
      }
    }
  }

  public static void main(String []args) throws Exception{
    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

    String[] tmp = br.readLine().split(" ");
    start = Integer.parseInt(tmp[0]);
    end = Integer.parseInt(tmp[1]);

    visited = new int[100000+1];
    System.out.println(bfs());
  } 
}

File: baekjoon/bfs/1697/1697.py
import sys

def bfs():
  queue = [start]
  visited[start] = 0
  while True:
    pop = queue.pop(0)
    if pop == end: return visited[pop]
    if pop+1 <= 100000 and visited[pop+1] == 0:
      queue.append(pop+1)
      visited[pop+1] = visited[pop] + 1
    if pop-1 >= 0 and visited[pop-1] == 0:
      queue.append(pop-1)
      visited[pop-1] = visited[pop] + 1
    if pop*2 <= 100000 and visited[pop*2] == 0:
      queue.append(pop*2)
      visited[pop*2] = visited[pop] + 1

start, end = list(map(int, sys.stdin.readline()[:-1].split(" ")))
visited = [0 for _ in range(100000+1)]

print(bfs())
